Clean up readme

ajweeks opened the pull request:
Several small changes, mostly to improve understandably
Capitalize new sentences

ajweeks committed:
Clean up readme

Several small changes, mostly to improve understandability
Capitalize new sentances

RoestVrijStaal reviewed (approved):
Please also mention in commit message that you stripped the file from trailing whitespaces.

ajweeks commented on README.md:
```diff
@@ -112,47 +109,49 @@ the DDS cubemap face order is pre-defined as SOIL_DDS_CUBEMAP_FACE_ORDER
     );
 
-load an image as a heightmap, forcing greyscale (so channels should be 1)
+Load an image as a heightmap, forcing greyscale (channels will be set to 1 on successful load)
```
It looks like SOIL.h incorrectly states that `SOIL_load_image` will return 0 on failure and 1 on success. In reality `NULL` is returned on failure and a valid pointer is returned on success. I will change this in the next commit

ajweeks committed:
Clean up readme & docs

Several functions stated that they returned 0 or 1,
when in reality they return a `const char*`. This has been
fixed.

One line in the readme was also made to be more correct.

The previous commit also stripped trailing whitespace
from the readme.
